python/lsst/ctrl/bps/panda/utils.py

```diff
@@ -93,16 +93,13 @@ def copy_files_for_distribution(files_to_stage, file_distribution_uri, max_copy_
     for local_pfn in files_to_stage.values():
         folder_name = os.path.basename(os.path.normpath(local_pfn))
         if os.path.isdir(local_pfn):
+            folder_uri = file_distribution_uri.join(folder_name, forceDirectory=True)
             files_in_folder = ResourcePath.findFileResources([local_pfn])
             for file in files_in_folder:
                 file_name = file.basename()
-                files_to_copy[file] = ResourcePath(
-                    os.path.join(file_distribution_uri, folder_name, file_name)
-                )
+                files_to_copy[file] = folder_uri.join(file_name)
         else:
-            files_to_copy[ResourcePath(local_pfn)] = ResourcePath(
-                os.path.join(file_distribution_uri, folder_name)
-            )
+            files_to_copy[ResourcePath(local_pfn)] = file_distribution_uri.join(folder_name)
```

Comment:
We know `local_pfn` is a file so we can say `ResourcePath(local_pfn, forceDirectory=False)` here.

I think the `file_distribution_uri.join(folder_name)` is the same as `folder_uri` above so maybe move the `folder_uri` outside the `if` statement and reuse the variable here.

Reply:
To clarify the slack discussion. `folder_name` is not a directory if `local_pfn` is not a directory so this should be:

```suggestion
            files_to_copy[ResourcePath(local_pfn, forceDirectory=False)] = file_distribution_uri.join(folder_name, forceDirectory=False)
```